File: 82.py
import numpy as np
import sys

vocab = {}

D = 5
with open("81.jl.out") as f:
    for line in f.readlines()[:1]:
        words = line[:-1].strip().split(" ")
        words = list(filter(lambda x: x != "", words))
        for idx in range(len(words)):
            randd = np.random.randint(1, D + 1)
            d = min(randd, idx + 1, len(words) - randd - 1)
            print(d)
            sys.stdout.write("{}\t".format(words[idx]))
            for i in range(-d, d + 1):
                if i != 0:
                    if idx + i < 0:
                        sys.stdout.write("N/A\t")
                    elif idx + i >= len(words):
                        sys.stdout.write("N/A\t")
                    else:
                        sys.stdout.write("{}\t".format(words[idx + i]))
            sys.stdout.write("\n")

File: 82.jl
D = 5

open("81.jl.out") do f
  for line in readlines(f)
    words = filter(x-> !isempty(x), strip.(split(line)))
    nw = length(words)
    for idx in 1:length(words)
      randd = rand(1:D)
      d = min(D, max(idx-1,1), nw-idx)
      print(words[idx], "\t")
      for i in -d:d
        i == 0 && continue
        if idx+i < 1 || idx+i > nw
          print("N/A\t")
        else
          print(words[idx+i], "\t")
        end
      end
      println()
    end
  end
end
